### ISSUE TITLE
Nuget pack doesnt pack assets that are private assets=all

### ISSUE THREAD

**Issue by ronnyek** [NuGet/Home#5455]
## tl;dr; version 
I just want some way to easily specify private dependencies so that consumers of my library don't have to go reach out to all the servers and fetch my libraries dependencies. Not the normal use case, but I think perfectly justifiable.

## Full Version
I am sure this is a misunderstanding on my part, but documentation seems to be severly lacking. I'm at my whits end and don't know who to ask at this point.

I've got a library that I am building that encapsulates another package for the purposes of enforcing proper usage, as well as abstracting these components.

MyInjection consomes Autofac. I don't care if a person explicitly adds autofac to their project on their own, that's fine. What I want is for MyInjection to wrap autofac. As it appears to be now, I import my library package and autofac gets fetched and installed as well. This means if we were talking about something other than autofac where the consumed library lived on another server... I've got to propagate nuget configs everywhere I want to be able to consume.

Given I don't think this is an common appropriate use case, I'd like to be able to say this library consumes autofac, but effectively just includes those specific versions of the dll (or some other way of merging) into the library itself.

If I set PrivateAssets="All" on the package reference, I do infact see it as no-longer appearing in the dependencies for the nuget package, however the actual library still references it. The assembles are not included in the nuget package at all, at this point.

So my question... what is the actual point? mark PrivateAssets="All" and the dependency disappears, and then we hope and prey the library exists through other means? It seems like the only real way to achieve what I want is to generate a nuspec, somehow try to intelligently parse the csproj to determine private assets dependency graph, and suck all those assemblies in from the bin folder (and even then I'm not sure if it'd work).

I feel like ultimately what is going to happen, I'm going to end up with hard coded dll's in source control because this stuff doesn't appear to work, or was never intended to work.

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe):
dotnet.exe version 1.0.4
visual studio 2017

NuGet version (x.x.x.xxx):

dotnet.exe : 1.0.4

VS version (if appropriate): 2017

OS version (i.e. win10 v1607 (14393.321)): windows 10

Worked before? If so, with which NuGet version:

## Detailed repro steps so we can see the same problem

1. dotnet new hwapp

2. add a package dependency on lib of your choice (Newtonsoft.Json for example)

3. Edit csproj, mark PrivateAssets="All" on the package reference

4. dotnet pack

5. Observe how the resulting nuget package neither depends on Newtonsoft.Json, nor does it include an actual distribution of newtonsoft.json.
...

**Comment by rohit21agrawal** [comment]
this is expected behavior since by marking PrivateAssets=all , you are essentially saying it's a build time dependency.

**Comment by michael-hawker** [comment]
- See now: #10375
